Remotion/Web/Development.WebTesting.UnitTests/Accessibility/AccessibilityAnalyzerTest.cs

```diff
@@ -0,0 +1,273 @@
+
+      Assert.That (analyzer.IncludeIFrames, Is.True);
+      Assert.That (analyzer.IFrameTimeout, Is.EqualTo (TimeSpan.FromSeconds (5)));
+    }
+
+    [Test]
+    public void CreateAnalyzer_ChangesDefaultConfiguration ()
+    {
+      _configStub
+          .Stub (_ => _.IFrameTimeout)
+          .Return (TimeSpan.FromSeconds (5));
+      _configStub
+          .Stub (_ => _.IncludeIFrames)
+          .Return (true);
+
+      var analyzer = new TestableAccessibilityAnalyzer (
+          _webDriverStub,
+          _jsExecutorStub,
+          _resultParserStub,
+          _configStub,
+          _sourceProviderStub,
+          _resultMapperStub,
+          _loggerStub);
+
+      analyzer.IFrameTimeout = TimeSpan.FromSeconds (15);
+      analyzer.IncludeIFrames = false;
+
+      Assert.That (analyzer.IncludeIFrames, Is.False);
+      Assert.That (analyzer.IFrameTimeout, Is.EqualTo (TimeSpan.FromSeconds (15)));
+    }
+
+    [Test]
+    public void Analyze_EmptyId_ThrowsException ()
+    {
+      var analyzer = new TestableAccessibilityAnalyzer (
+          _webDriverStub,
+          _jsExecutorStub,
+          _resultParserStub,
+          _configStub,
+          _sourceProviderStub,
+          _resultMapperStub,
+          _loggerStub);
+
+      Assert.That (() => analyzer.Analyze (""), Throws.ArgumentException);
+    }
+
+    [Test]
+    public void Analyze_ExecuteAsyncReturnsNull_ThrowsInvalidOperationException ()
+    {
+      _jsExecutorStub
+          .Stub (_ => _.ExecuteAsyncScript (null, null))
+          .IgnoreArguments()
+          .Return (null);
+      _jsExecutorStub
+          .Stub (_ => _.ExecuteScript ("return (typeof axe !== 'undefined')"))
+          .Return (true);
+      _jsExecutorStub
+          .Stub (_ => _.ExecuteScript ("return self.name;"))
+          .Return ("TestFrame");
+      _webDriverStub
+          .Stub (_ => _.FindElements (null))
+          .IgnoreArguments()
+          .Return (new List<IWebElement>().AsReadOnly());
+      _webDriverStub.Stub (_ => _.SwitchTo().DefaultContent());
+      var analyzer = new TestableAccessibilityAnalyzer (
+          _webDriverStub,
+          _jsExecutorStub,
+          _resultParserStub,
+          _configStub,
+          _sourceProviderStub,
+          _resultMapperStub,
+          _loggerStub);
+
+      Assert.That (
+          () => analyzer.Analyze(),
+          Throws.TypeOf (typeof (InvalidOperationException)).With.Message.EqualTo ("Could not obtain accessibility analysis result."));
+    }
+
+    [Test]
+    public void Analyze_IEPath ()
+    {
+      var axeResult = new AxeResult();
+      var accessibilityResult = new AccessibilityResult (
+          DateTime.MinValue,
+          "url",
+          "version",
+          0,
+          0,
+          0,
+          "orientationType",
+          "userAgent",
+          true,
+          AccessibilityConformanceLevel.Wcag20_ConformanceLevelDoubleA,
+          new List<AccessibilityRuleResult>());
+      _jsExecutorStub
```

Comment:
can use single line if there's enough line width available. linewrap only if there's long statements or expected to become long statements.